src/start.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { getEnvBoolean } from './utils';
+import { startMetrics, MetricsOptions } from './metrics';
+import { startProfiling, ProfilingOptions } from './profiling';
+import { startTracing, TracingOptions } from './tracing';
+
+interface GeneralOptions {
+  accessToken: string;
+  /**
+   * This is intentionally optional so that the exporters would be able to enforce
+   * their own default.
+   */
+  endpoint?: string;
+  serviceName: string;
+}
+
+interface Options extends GeneralOptions {
+  metrics: MetricsOptions;
+  profiling: ProfilingOptions;
+  tracing: TracingOptions;
```

Comment:
If one were to enable metrics and profiling, `start` would look like:
```typescript
start({ profiling: {}, metrics: {} });
```

Would it make more sense if it's either boolean or the configuration explicitly in types? :thinking:

Reply:
That was the intention. It's implemented as such in the logic but forgot the types. Thanks!